core/src/test/java/com/squarespace/template/plugins/platform/ContentFormattersTest.java

```diff
@@ -229,4 +229,30 @@ public void testWidthHeight() throws CodeException {
     assertFormatter(HEIGHT, json, "200");
   }
 
+  @Test
+  public void testWebsiteColor() throws CodeException {
+    String template = "{valid-hsla|website-color}";
+    String validHslaJson = "{\"valid-hsla\": {\"hue\": 200, \"saturation\": 0.3, \"lightness\": 0.5, \"alpha\": 0.9}}";
+    Instruction code = compiler().compile(template).code();
+    Context ctx = compiler().newExecutor().code(code).json(validHslaJson).execute();
+    String result = eval(ctx);
+    assertEquals(result, "hsla(200, 30%, 50%, 0.9)");
+
+    template = "{valid-hsl|website-color}";
+    String validHslJson = "{\"valid-hsl\": {\"hue\": 110, \"saturation\": 0.9, \"lightness\": 1}}";
+    code = compiler().compile(template).code();
+    ctx = compiler().newExecutor().code(code).json(validHslJson).execute();
+    result = eval(ctx);
+    assertEquals(result, "hsl(110, 90%, 100%)");
+
+    template = "{invalid-hsla|website-color}";
+    String invalidHslaJson = "{\"invalid-hsla\": {\"hue\": 400, \"saturation\": -0.3, \"lightness\": 5, \"alpha\": 3}}";
+    code = compiler().compile(template).code();
+    ctx = compiler().newExecutor().code(code).json(invalidHslaJson).execute();
+    result = eval(ctx);
+    assertEquals(result, "Hue value is out of bounds. It should be between 0 and 360. "
```

Comment:
What's the intended behavior if the formatter is pointed at an object where h/s/l/a are not numbers? Or where one of the properties is missing?

Reply:
if one of the (non-alpha) properties is missing, it will be evaluated to 0.  if it's not a number, it will be converted either to 0 or to its associated string/boolean number

should there be additional error messaging and validation? i wasn't sure how much validation i should be including in this formatter, since the other formatters in this compiler don't do much validation and seem to assume that the arguments are mostly correct

Reply:
@phensley is better suited to weigh in on this since it's more of a question of language design. I think we should ideally be implementing extensive validation rules in business logic rather than in JSON-T. It is a fairly common mistake to point a formatter at the wrong part of render context, or at a property that doesn't exist, so printing an error rather than simply outputting 0 may help from a dev ex standpoint

Reply:
The simpler formatters that have very little logic typically have concise error handling.